test/automation/src/positron/positronPopups.ts

```diff
@@ -76,4 +76,25 @@ export class PositronPopups {
 		const toastLocator = this.code.driver.getLocator(NOTIFICATION_TOAST);
 		await toastLocator.waitFor({ state: 'attached', timeout: 20000 });
 	}
+
+	async waitForModalDialogBox() {
+		await this.code.waitForElement(POSITRON_MODAL_DIALOG_BOX);
+	}
```

Comment:
Just noting that I am adding `popupCurrentlyOpen`, which is related to this, but is for checking if there is a popup currently open, rather than waiting for an expected modal dialog.

Waiting is helpful when we expect to see a modal dialog before doing some action. Checking if a popup is currently open is helpful if we have behaviour conditional on if a popup has been opened or not.

I may borrow your technique below in `clickOnModalDialogPopupOption` to add the ability to check if a specific popup is currently open based on the modal title 😄